File: ViewVb/Models/MySampleModel.vb
''  -*-  coding: utf-8-with-signature  -*-  ''
''************************************************************************
''                                                                      ''
''                  ---   Graphics Test Project.   ---                  ''
''                                                                      ''
''          Copyright (C), 2025-2026, Takahiro Itou                     ''
''          All Rights Reserved.                                        ''
''                                                                      ''
''          License: (See COPYING or LICENSE files)                     ''
''          GNU Affero General Public License (AGPL) version 3,         ''
''          or (at your option) any later version.                      ''
''                                                                      ''
''************************************************************************

Imports FullColorImage = SampleWrapper.Images.FullColorImage


Namespace Global.ViewVb.Models

Public Class MySampleModel

''======================================================================
''
''    Member Variables.
''

Private  ReadOnly   m_imgBuffer As FullColorImage

Private  ReadOnly   m_imgWidth  As Integer
Private  ReadOnly   m_imgHeight As Integer


''======================================================================
''
''    Constructor(s) and Destructor.
''

Public Sub New(
        ByVal nWidth As Integer,
        ByVal nHeight As Integer,
        ByVal cbPixel As Integer,
        ByVal lStride As Integer)
''--------------------------------------------------------------------
''    コンストラクタ
''--------------------------------------------------------------------
    m_imgWidth  = nWidth
    m_imgHeight = nHeight

    m_imgBuffer = New FullColorImage()
    Me.m_imgBuffer.allocateImage(nWidth, nHeight, cbPixel, lStride)
End Sub


''======================================================================
''
''    Public Member Functions.
''

Public Overridable Sub clearImage(ByVal colBG As Integer)
''--------------------------------------------------------------------
''    画像をクリアする。
''--------------------------------------------------------------------
    Me.m_imgBuffer.fillRectangle(0, 0, Me.m_imgWidth, Me.m_imgHeight, colBG)
End Sub


Public Overridable Sub drawSampleImage()
''--------------------------------------------------------------------
''    サンプル画像を描画する。
''--------------------------------------------------------------------
Dim colBG As Integer
Dim colTL As Integer
Dim colTR As Integer
Dim colBL As Integer
Dim colBR As Integer
Dim rnd As New Random()

    ' 色を適当に決める。背景はある程度明るい色
    colBG = rnd.Next(16777216) Or &HFF808080

    ' 色を適当に決める。
    colTL = rnd.Next(256) Or &HFF000080
    colTR = (rnd.Next(256) * 256) OR &HFF008000
    colBL = rnd.Next(256)
    colBL = (colBL * 257) Or &HFF008080
    colBR = (rnd.Next(256) * 65536) OR &HFF800000

    Me.m_imgBuffer.drawSample(colBG, colTL, colTR, colBL, colBR)
End Sub


''======================================================================
''
''    Properties.
''

''--------------------------------------------------------------------
''    イメージバッファを取得するプロパティ。
''--------------------------------------------------------------------
Public  ReadOnly  Property  ImageBuffer() As FullColorImage
    Get
        Return  Me.m_imgBuffer
    End Get
End Property


End Class

End Namespace

File: ViewCs/Models/MySampleModel.cs
//  -*-  coding: utf-8-with-signature;  mode: c++  -*-  //
/*************************************************************************
**                                                                      **
**                  ---   Graphics Test Project.   ---                  **
**                                                                      **
**          Copyright (C), 2025-2026, Takahiro Itou                     **
**          All Rights Reserved.                                        **
**                                                                      **
**          License: (See COPYING or LICENSE files)                     **
**          GNU Affero General Public License (AGPL) version 3,         **
**          or (at your option) any later version.                      **
**                                                                      **
*************************************************************************/

using FullColorImage = SampleWrapper.Images.FullColorImage;


namespace  ViewCs.Models  {

public  class  MySampleModel
{

//========================================================================
//
//    Constructor(s) and Destructor.
//

//----------------------------------------------------------------
/**   コンストラクタ。
**
**/
public
MySampleModel(
        int nWidth,
        int nHeight,
        int cbPixel,
        int lStride)
{
    m_imgWidth  = nWidth;
    m_imgWidth  = nHeight;
    m_imgBuffer = new FullColorImage();
    this.m_imgBuffer.allocateImage(nWidth, nHeight, cbPixel, lStride);
}


//========================================================================
//
//    Public Member Functions.
//

//----------------------------------------------------------------
/**   画像をクリアする。
**
**/
public  virtual  void
clearImage(int colBG)
{
    this.m_imgBuffer.fillRectangle(
            0, 0, this.m_imgWidth, this.m_imgHeight, colBG);
}

//----------------------------------------------------------------
/**   サンプル画像を描画する。
**
**/
public  virtual  void
drawSampleImage()
{
    int     cAlpha;
    int     colBG, colTL, colTR, colBL, colBR;
    System.Random   rnd = new System.Random();

    //  色を適当に決める。背景はある程度明るい色
    cAlpha  = 255 << 24;
    colBG = rnd.Next(16777216) | cAlpha | 0x00808080;

    //  色を適当に決める。
    colTL = rnd.Next(256) | cAlpha | 0x00000080;
    colTR = (rnd.Next(256) <<  8) | cAlpha | 0x00008080;
    colBL = rnd.Next(256);
    colBL = (colBL | colBL <<  8) | cAlpha | 0x00008080;
    colBR = (rnd.Next(256) << 16) | cAlpha | 0x00800000;

    this.m_imgBuffer.drawSample(colBG, colTL, colTR, colBL, colBR);
}


//========================================================================
//
//    Properties.
//

//----------------------------------------------------------------
/**   イメージバッファを取得するプロパティ。
**
**/

public  FullColorImage
ImageBuffer {
    get { return  this.m_imgBuffer; }
}


//========================================================================
//
//    Member Variables.
//

private  readonly   FullColorImage  m_imgBuffer;

private  readonly   int             m_imgWidth;
private  readonly   int             m_imgHeight;


}   //  End class  MySampleModel

}   //  End of namespace  ViewCs.Models
